includes/class-events.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Automattic\WP\Cron_Control;
+
+class Events extends Singleton {
+	/**
+	 * PLUGIN SETUP
+	 */
+
+	/**
+	 * Class properties
+	 */
+	const LOCK = 'run-events';
+
+	/**
+	 * Register hooks
+	 */
+	protected function class_init() {
+		// Prime lock cache if not present
+		Lock::prime_lock( self::LOCK );
+	}
+
+	/**
+	 * List events pending for the current period
+	 */
+	public function get_events() {
+		$events = get_option( 'cron' );
+
+		// That was easy
+		if ( ! is_array( $events ) || empty( $events ) ) {
+			return array( 'events' => null, );
+		}
+
+		// To be safe, re-sort the array just as Core does when events are scheduled
+		// Ensures events are sorted chronologically
+		uksort( $events, 'strnatcasecmp' );
```

Comment:
Core only re-orders the array when `wp_schedule_event()` is called, which probably happens frequently enough (`wp_reschedule_event()` calls it too), but I included this here just in case something directly manipulated the option. Since Core doesn't reorder when calling `_get_cron_array()`, it's probably safe for me to remove this. :)